rest/model/codegen.go

```diff
@@ -63,84 +63,113 @@ type modelConversionInfo struct {
 		Input: schema,
 	}
 	fileTemplate, err := getTemplate(fileTemplatePath)
 	if err != nil {
-		return nil, errors.Wrap(err, "error loading file template")
+		return nil, nil, errors.Wrap(err, "error loading file template")
 	}
 	structTemplate, err := getTemplate(structTemplatePath)
 	if err != nil {
-		return nil, errors.Wrap(err, "error loading struct template")
+		return nil, nil, errors.Wrap(err, "error loading struct template")
 	}
 	fieldTemplate, err := getTemplate(fieldTemplatePath)
 	if err != nil {
-		return nil, errors.Wrap(err, "error loading field template")
+		return nil, nil, errors.Wrap(err, "error loading field template")
 	}
 	parsedAst, parseErr := gqlparser.LoadSchema(&source)
 	if parseErr != nil {
-		return nil, errors.Wrap(parseErr, "error parsing schema")
+		return nil, nil, errors.Wrap(parseErr, "error parsing schema")
 	}
 	catcher := grip.NewBasicCatcher()
 	structs := ""
-	conversionCode := ""
+	serviceCode := ""
+	generatedConversions := map[string]string{}
 
 	typeNames := []string{}
 	for key := range parsedAst.Types {
 		typeNames = append(typeNames, key)
 	}
 	sort.Strings(typeNames)
 	for _, typeName := range typeNames {
-		dbModel, shouldConvert := config[typeName]
-		if !shouldConvert {
+		gqlType := parsedAst.Types[typeName]
+		if gqlType == nil {
+			catcher.Errorf("unable to find type '%s'", typeName)
 			continue
 		}
-		gqlType := parsedAst.Types[typeName]
 		if gqlType.BuiltIn {
 			continue
 		}
 		if gqlType.Kind == ast.Scalar {
 			continue
 		}
-		fields := ""
-		extractedFields := extractedFields{}
-		for _, field := range gqlType.Fields {
-			fieldInfo := getFieldInfo(field)
-			extractedFields[field.Name] = fieldInfo
-			var fieldData string
-			fieldData, err = output(fieldTemplate, fieldInfo)
-			if err != nil {
-				catcher.Add(err)
-				continue
-			}
-			fields += fieldData
+		dbModel, shouldConvert := config[typeName]
+		if !shouldConvert {
+			continue
 		}
-		var structData string
-		structData, err = output(structTemplate, structInfo{Name: typeName, Fields: fields})
+
+		structData, code, err := generateForStruct(fieldTemplate, structTemplate, typeName, dbModel, *gqlType, generatedConversions)
 		if err != nil {
 			catcher.Add(err)
 			continue
 		}
 		structs += structData
+		serviceCode += code
+	}
+	modelFile, err := output(fileTemplate, fileInfo{Package: "model", Structs: structs, Code: serviceCode})
+	catcher.Add(err)
+	formattedModelCode, err := goimports(modelFile)
+	catcher.Add(err)
 
-		parts := strings.Split(dbModel, ".")
-		if len(parts) < 2 {
-			return nil, errors.Errorf("invalid format for DB model: %s", dbModel)
-		}
-		code, err := createConversionMethods(strings.Join(parts[:len(parts)-1], "."), parts[len(parts)-1], extractedFields)
-		if err != nil {
-			return nil, errors.Wrapf(err, "error generating conversion methods for type '%s'", typeName)
-		}
-		conversionCode += code
+	converters := []string{}
+	conversionCode := ""
+	for key := range generatedConversions {
```

Comment:
You've done this block a couple times, maybe a util "GetSortedKeys" would be more readable

Reply:
I tried that previously, but it'll require multiple functions with different signatures since the inputs are map[string]Something so it won't make things cleaner